guide/session.md

```diff
@@ -103,32 +106,32 @@ Things to remember:
    	return next();
    });
 
-   // incremented count will still be written to store
+   // Incremented count will still be written to store even though there is an uncaught error in the previous middleware
 
    ```
 
-4. If you use webhook mode, you must disable `webhookReply`.
+4. If your bot is on webhook mode, you must disable `webhookReply`.
 
    ```TS (Node)
    const bot = new Telegraf(token, { telegram: { webhookReply: false } })
    ```
 
-5. You must not introduce race-conditions of your own. Be wary of writing stale data back to session. Example:
+5. You must not introduce race-conditions of your own. Be wary of writing stale data to the session store:
 
    ```TS (Node)
    bot.on(message("text"), async ctx => {
-   	// reading value from session
+   	// Reading value from session
    	const count = ctx.session.count;
 
    	const response = await fetch(API_URL, { body: count });
 
-   	// ⚠️ WARNING! You wrote stale value to session.
-   	// Another request may have updated session while you awaited fetch, be careful of that
+   	// ⚠️ WARNING! You are writing stale value to the session.
+   	// Another request may have updated this session property while you awaited the fetch request, be careful of additional or unwanted changes
    	ctx.session.count = count + 1;
 
-   	// ✅ Good, you wrote immediately after reading
+   	// ✅ Good, you are changing the value based on a immediate reading
    	ctx.session.count = ctx.session.count + 1;
    });
    ```
 
-##### Note: Telegraf's session safety will not currently work with long-polling using NodeJS clusters or other means of threading (workers, etc.). It will however work fine with single-instance long-polling, or multi-instance webhook mode.
+##### Note: Telegraf's session safety does not currently work with long-polling mode using NodeJS clusters or other means of threading (workers, etc.). It will however work fine with single-instance long-polling, or multi-instance webhook mode.
```

Comment:
It will not work. It's a strong assertion. There was no need to rephrase.

Reply:
I read all the comments, thanks for taking the time and pointing out my mistakes.

The broadcasting or formatting file was causing issues when cloning the repo, and when i downloaded the ZIP and pasted them separately, windows automatically changed their name to ___. ... So I decided it would be better just to have them there as their default name until the maintainers come up with a solution of where they should put them. I'm not really proficient at git but i guess it would cause conflicts if i just didn't include those files at all or it outright would count as a deletion?

And yes sometimes i just included more words but it was mainly because i had free time on my hand and thought it would be beneficial to have a little more clarity (for example i thought the Typescript one would be beneficial to a new TS user so they know exactly the purpose of augmenting the ctx type).

Either way i will try to make more useful PRs; again thanks for taking the time and pointing out the mistakes.




> It will not work. It's a strong assertion. There was no need to rephrase.

Reply:
> for example i thought the Typescript one

Sure, that's a minor improvement. Most changes are not improvements, that's the problem.

Please do make smaller and more meaningful PRs. Thanks!

Reply:
Try to use `git` though, avoid downloading as zip. It's easy:

1. Fork the project
2. Clone your git repo:

	```TS
	git clone https://github.com/username/repo.git
	```

3. Make changes
4. Make a commit

	```TS
	git commit -m "commit message"
	```

5. Push changes to your repo. This might ask for a password, for which you need to have generated a Personal Access Token (https://github.com/settings/tokens/new)

	```TS
	git push -u origin master
	```

6. Open your repo on GitHub, and make a PR

Reply:
Thanks for helping out, previously I did all these steps just fine but the issue is with the broadcasting and formatting files, which will throw an error saying: Invalid path '???" etc.
So I just forked the repo with SSH, then downloaded the ZIP and pasted those two files separately but windows changed their name to ___... . is it fine if I not include them when I make a commit or will it count as a deletion?